datasets/openpi/openpi.py

```diff
@@ -0,0 +1,247 @@
+
+from __future__ import absolute_import, division, print_function
+
+import json
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{tandon-etal-2020-dataset,
+    title = "A Dataset for Tracking Entities in Open Domain Procedural Text",
+    author = "Tandon, Niket  and
+      Sakaguchi, Keisuke  and
+      Dalvi, Bhavana  and
+      Rajagopal, Dheeraj  and
+      Clark, Peter  and
+      Guerquin, Michal  and
+      Richardson, Kyle  and
+      Hovy, Eduard",
+    booktitle = "Proceedings of the 2020 Conference on Empirical Methods in Natural Language Processing (EMNLP)",
+    month = nov,
+    year = "2020",
+    address = "Online",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/2020.emnlp-main.520",
+    doi = "10.18653/v1/2020.emnlp-main.520",
+    pages = "6408--6417",
+}
+"""
+
+# TODO: Add description of the dataset here
+# You can copy an official description
+_DESCRIPTION = """\
+OpenPI dataset for tracking entities in open domain procedural text.
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = "https://allenai.org/data/openpi"
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = "MIT License"
+
+# TODO: Add link to the official dataset URLs here
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_BASE_URL = "https://raw.githubusercontent.com/allenai/openpi-dataset/main/data/gold/"
+
+
+class OpenpiConfig(datasets.BuilderConfig):
+    """"BuilderConfig for OpenPI Dataset"""
+
+    def __init__(self, mode, type_, **kwargs):
+        super(OpenpiConfig, self).__init__(**kwargs)
+        self.mode = mode
+        self.type_ = type_
+
+
+class Openpi(datasets.GeneratorBasedBuilder):
+    """A Dataset for Tracking Entities in Open Domain Procedural Text"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIG_CLASS = OpenpiConfig
+
+    BUILDER_CONFIGS = [
+        OpenpiConfig(
+            name="questions",
+            mode="question",
+            type_="data",
+            version=VERSION,
+            description="Provides the question sentences",
+        ),
+        OpenpiConfig(
+            name="questions-metadata",
+            mode="question",
+            type_="metadata",
+            version=VERSION,
+            description="Provides the question metadata",
+        ),
+        OpenpiConfig(
+            name="answers",
+            mode="answers",
+            type_="data",
+            version=VERSION,
+            description="Provides the answer sentences",
+        ),
+        OpenpiConfig(
+            name="answers-metadata",
+            mode="answers",
+            type_="metadata",
+            version=VERSION,
+            description="Provides the answer metadata",
+        ),
+    ]
```

Comment:
Do you think it would be useful to have only one config with all the data in it ?
Not sure if it makes sense to have a config with only the question and no metadata nor answers + answers metadata

Reply:
The Data are split across 4 files tied together only with a single ID field and the files are split that way (Given in https://github.com/Bharat123rox/datasets/blob/6a1b6dd23269c62eadfb4a53ffb0b8ba83a149c7/datasets/openpi/openpi.py#L188-L193)
(Metadata provide additional information different from the raw questions/answers, and all of them can be "joined" or tied only with the ID of the question, so I felt that the user could fetch each of these aspects on demand, hence the 4-way config)

If not, how do you visualize the ideal data model? Also, if there is a way where we could specify some sort of joining key, I'm definitely open to making these changes!